src/manager.cpp

```diff
@@ -793,5 +793,39 @@ void Manager::hostStateChangeCallBack(sdbusplus::message_t& i_msg)
     }
 }
 
-void Manager::performVPDRecollection() {}
+void Manager::performVpdRecollection()
+{
+    try
+    {
+        if (m_worker.get() != nullptr)
+        {
+            nlohmann::json l_sysCfgJsonObj = m_worker->getSysCfgJsonObj();
+
+            // Check if system config JSON is present
+            if (l_sysCfgJsonObj.empty())
+            {
+                throw std::runtime_error(
+                    "System config json object is empty, can't process recollection.");
+            }
+
+            const auto& l_frusReplaceableAtStandby =
+                jsonUtility::getListOfFrusReplaceableAtStandby(l_sysCfgJsonObj);
+
+            for (const auto& l_fruInventoryPath : l_frusReplaceableAtStandby)
+            {
+                collectSingleFruVpd(l_fruInventoryPath);
```

Comment:
How do we know the recollection failures and for which ? is the called code's error log good enough?

Reply:
Yup, "collectSingleFruVpd" will log the required logs in case of any error.